src/main/java/witchinggadgets/common/blocks/tiles/TileEntityLabelLibrary.java

```diff
@@ -133,7 +133,19 @@ public void openInventory() {}
     public void closeInventory() {}
 
     @Override
-    public boolean isItemValidForSlot(int i, ItemStack itemstack) {
-        return true;
+    public boolean isItemValidForSlot(int slot, ItemStack itemstack) {
+        return slot == 0 && isLabel(itemstack) && clearLabelAspects(itemstack);
+    }
+
+    public boolean isLabel(ItemStack stack) {
+        return stack != null && stack.getItem().equals(ConfigItems.itemResource) && stack.getItemDamage() == 13;
```

Comment:
`Item` doesn't have any special `.equals`.
```suggestion
        return stack != null && stack.getItem() == ConfigItems.itemResource && stack.getItemDamage() == 13;
```

Reply:
was the original code used in the container but fair I guess